File: o15/o15.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
    let count=0;
    while (n>1) {
        if (n&1==1) {
            count+=1;
        }
        n=n>>>1;
    }
    if (n==1) {
        count+=1;
    }
    return count;
};

let n=0b11111111111111111111111111111101;
console.log(hammingWeight(n));

File: o15/o15.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        #print(bin(n)[2:].zfill(32))
        count = 0
        while n > 1:
            if n & 1 == 1:
                count += 1
            n = n >> 1
        if n == 1:
            count += 1
        return count


if __name__ == "__main__":
    n = 128
    s = Solution()
    print(s.hammingWeight(n))